### PR TITLE
Debug mode keys when `debug` mode is on

### PR DESCRIPTION
**Pull request by stefanhendriks**
for - #188

This also makes most code that uses `key[<SCANCODE>]` react on the keyboard events.  The exceptions are on units or other places where we remove the code or defer to later. 

To test: start game without `-debug` flag and you should not be able to use debug keys (`TAB`- etc). 

- most obvious things player noticed where changing player (`TAB-<nr>`)

### CODE REVIEW

File: building/cItemBuilder.cpp [hunk 1]
```diff
@@ -758,3 +759,32 @@ void cItemBuilder::removeItemsByBuildId(eBuildType buildType, int buildId) {
...
+            onKeyPressed(event);
+            break;
+        default:
+            break;
+    }
+}
+
+void cItemBuilder::onKeyHold(const cKeyboardEvent &event) {
+    if (event.hasEitherKey(KEY_LSHIFT, KEY_RSHIFT)) {
+        m_buildItemMultiplierEnabled = true;
```
Comment (pbomta) [inline building/cItemBuilder.cpp:781]: Nice. Can this class now get rid of the allegro keyboard include?

File: drawers/cBuildingListDrawer.h [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 
-class cBuildingListDrawer {
+#include "observers/cInputObserver.h"
+
+class cBuildingListDrawer : cInputObserver {
 
 public:
-	cBuildingListDrawer(cPlayer *thePlayer);
-	~cBuildingListDrawer();
+	explicit cBuildingListDrawer(cPlayer *player);
```
Comment (pbomta) [inline drawers/cBuildingListDrawer.h:8]: 👍

File: tools/LogViewer/nbproject/build-impl.xml [hunk 3]
```diff
@@ -491,7 +491,7 @@ is divided into following sections:
     </target>
     <!--
                 =================
-                DEBUGGING SECTION
+                game.isDebugMode() SECTION
```
Comment (pbomta) [inline tools/LogViewer/nbproject/build-impl.xml:494]: this was probably not meant to be renamed :) also other .xml files

File: utils/cLog.h [hunk 4]
```diff
@@ -62,5 +64,7 @@ class cLogger {
 
       long startTime; // start time of logging in miliseconds
 
+      bool m_debugMode;
```
Comment (pbomta) [inline utils/cLog.h:67]: Eventually, we could let the debug mode be dependent on the log level, but that's not easy to do half-half. We'll need to decide that and change all the calls at once, I expect.

**Review by pbomta — APPROVED** [review]

File: controls/cKeyboardEvent.h [hunk 5]
```diff
@@ -32,10 +32,9 @@ class cKeyboardEvent {
         str.append(partOne);
         str.append("[");
         for (auto key : event.keys) {
-            str.append(std::to_string(key));
-            str.append("=");
-            str.append(std::to_string(scancode_to_ascii(key)));
-            str.append(" ");
+            char msg[10];
+            sprintf(msg, "%s,", scancode_to_name(key));
```
[the remark this answers is not in the document]
Comment (stefanhendriks) [reply controls/cKeyboardEvent.h:36]: I don't know why this didn't work ( I tried `std.append` first). I'll retry later.

File: tools/LogViewer/nbproject/build-impl.xml (hunk 3 above)
Comment (stefanhendriks) [reply tools/LogViewer/nbproject/build-impl.xml:494]: 🤦‍♂️

File: utils/cLog.h (hunk 4 above)
Comment (stefanhendriks) [reply utils/cLog.h:67]: Hmm. Well, in the end `debugMode` basically enables some 'flags' to allow us to see (visually, like drawing on screen) different things. It *also* has impact on logging atm.

So those things need to be untangled:
- log levels (like you said)
- debug/drawing information per feature perhaps? (with some kind of variables)

For some other time for sure :)

File: controls/cKeyboardEvent.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (stefanhendriks) [reply controls/cKeyboardEvent.h]: fixed

**Pull request merged**